packages/model-generator/src/s3_string_object_fetcher.ts

```diff
@@ -14,15 +15,28 @@ export class S3StringObjectFetcher {
    */
   fetch = async (uri: string) => {
     const { bucket, key } = this.parseS3Uri(uri);
-    const getSchemaCommandResult = await this.s3Client.send(
-      new GetObjectCommand({ Bucket: bucket, Key: key })
-    );
-    const schema = await getSchemaCommandResult.Body?.transformToString();
-    if (!schema) {
-      // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
-      throw new Error('Error on parsing output schema');
+    try {
+      const getSchemaCommandResult = await this.s3Client.send(
+        new GetObjectCommand({ Bucket: bucket, Key: key })
+      );
+      const schema = await getSchemaCommandResult.Body?.transformToString();
+      if (!schema) {
+        // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
+        throw new Error('Error on parsing output schema');
+      }
+      return schema;
+    } catch (caught) {
+      if (caught instanceof NoSuchBucket) {
+        //eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
+        throw new AmplifyFault('NoSuchBucketFault', {
```

Comment:
should I throw this as an Error or as an AmplifyFault?

Reply:
`AmplifyFault` is correct here.